### PR TITLE
hv: ptdev: simplify ptdev_intx_pin_remap logic

### PR DESCRIPTION
**Pull request by lifeix**
Since this interrupt pin is passed through to the guest, we could set
it as we like. There's no need a flag to emulate the masked status.

Signed-off-by: Li, Fei1 <fei1.li@intel.com>

### CODE REVIEW

File: hypervisor/arch/x86/assign.c [hunk 1]
```diff
@@ -671,18 +671,14 @@ static void activate_physical_ioapic(struct vm *vm,
...
-	ioapic_set_rte(phys_irq, rte);
-
 	/* update irq trigger mode according to info in guest */
 	if ((rte.full & IOAPIC_RTE_TRGRMOD) == IOAPIC_RTE_TRGRLVL) {
 		is_lvl_trigger = true;
 	}
 	set_irq_trigger_mode(phys_irq, is_lvl_trigger);
 
-	/* enable interrupt */
-	GSI_UNMASK_IRQ(phys_irq);
```
Comment (anthonyzxu) [inline hypervisor/arch/x86/assign.c]: phys_irq is masked in this function, I think we need unmask here.

**Comment by lifeix** [comment]
@anthonyzxu @JasonChenCJ 
We can just simple the logic to call activate_physical_ioapic in ptdev_intx_pin_remap without change ptdev_intx_pin_remap. But if the guest mask the interrupt and the device could raise interrupt, our hypervisor would waste time to deal with this although we drop the interrupt at last.

**Comment by lifeix** [comment]
@NanlinXie @lijinxia 
Please help to have a test
Thanks.

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**